hyperspy/io_plugins/usid_hdf5.py

```diff
@@ -0,0 +1,447 @@
+    units = units.replace('[', '')
+    units = units.replace(']', '')
+    return quant, units
+
+
+def __convert_to_signal_dict(ndim_form, quantity, units, dim_dict_list,
+                             h5_path, h5_dset_path, sig_type=''):
+    """
+    Packages required components that make up a Signal object
+
+    Parameters
+    ----------
+    ndim_form : numpy.ndarray
+        N-dimensional form of the main dataset
+    quantity : str
+        Physical quantity of the measurement
+    units : str
+        Corresponding units
+    dim_dict_list : list
+        List of dictionaries that instruct the axes corresponding to the main dataset
+    h5_path : str
+        Absolute path of the original USID HDF5 file
+    h5_dset_path : str
+        Absolute path of the USIDataset within the HDF5 file
+    sig_type : str
+        Type of measurement
+
+    Returns
+    -------
+
+    """
+
+    sig = {'data': ndim_form,
+           'axes': dim_dict_list,
+           'metadata': {
+               'Signal':{'quantity': quantity, 'units': units, 'signal_type': sig_type},
+               'General': {'original_filename': h5_path, 'dataset_path': h5_dset_path,
+                           'original_file_type': 'USID HDF5', 'pyUSID_version': usid.__version__}
+           }}
+    return sig
+
+
+def __usidataset_to_signal(h5_main, verbose=False, ignore_non_linear_dims=True):
+    """
+    Converts a single specified USIDataset object to one or more Signal objects
+    Parameters
+    ----------
+    h5_main : pyUSID.USIDataset object
+        USID Main dataset
+    verbose : bool, Optional. Default = False
+        Whether or not to print debugging statements
+    ignore_non_linear_dims : bool, Optional
+        If True, parameters that were varied non-linearly in the desired dataset will result in Exceptions.
+        Else, all such non-linearly varied parameters will be treated as linearly varied parameters and
+        a Signal object will be generated.
+
+    Returns
+    -------
+    list of hyperspy.signals.BaseSignal objects. USIDatasets with compound datatypes are broken down to multiple
+    Signal objects.
+    """
+    h5_main = usid.USIDataset(h5_main)
+    # TODO: Cannot handle data without N-dimensional form
+    # First get dictionary of axes that HyperSpy likes to see. Ignore singular dimensions
+    pos_dict = __get_dim_dict(h5_main.pos_dim_labels,
+                              usid.hdf_utils.get_attr(h5_main.h5_pos_inds, 'units'),
+                              h5_main.get_pos_values,
+                              ignore_non_linear_dims=ignore_non_linear_dims)
+    spec_dict = __get_dim_dict(h5_main.spec_dim_labels,
+                               usid.hdf_utils.get_attr(h5_main.h5_spec_inds, 'units'),
+                               h5_main.get_spec_values,
+                               ignore_non_linear_dims=ignore_non_linear_dims)
+
+    num_spec_dims = len(spec_dict)
+    num_pos_dims = len(pos_dict)
+    if verbose:
```

Comment:
Use the logger instead of the `print` function?

Reply:
Done